### PR TITLE
add round trip tests

### LINKED ISSUE #11: Add round-trip test cases for all types that implement `(Try)FromJavaValue` and `(Try)IntoJavaValue`
Round-trip test: Java -> Rust -> Java

As an example, see [`test/driver/src/test(java(UserTest.java:boolTest(boolean)`](https://github.com/giovanniberti/robusta/blob/8ddb07e038e0c3d83e8793b8cbc6720e32213793/tests/driver/src/test/java/UserTest.java?_pjax=%23js-repo-pjax-container%2C%20div%5Bitemtype%3D%22http%3A%2F%2Fschema.org%2FSoftwareSourceCode%22%5D%20main%2C%20%5Bdata-pjax-container%5D#L22) and [also here for the Rust implementation](https://github.com/giovanniberti/robusta/blob/8ddb07e038e0c3d83e8793b8cbc6720e32213793/tests/driver/native/src/lib.rs?_pjax=%23js-repo-pjax-container%2C%20div%5Bitemtype%3D%22http%3A%2F%2Fschema.org%2FSoftwareSourceCode%22%5D%20main%2C%20%5Bdata-pjax-container%5D#L40).

All these tests can be very easily auto-generated if needed (at least from Rust with a declarative macro).

### PR DESCRIPTION
**Pull request by ahouts**
closes #11

Compiler complained about using `extern "jni"` with `macro_rules`. Maybe the macros were applied in the wrong order or something.

### CODE REVIEW

**Comment by giovanniberti** [comment]
Wonderful! Thank you for the PR 😃  

I'll investigate the error you get with Vec and get back to you

File: tests/driver/src/test/java/UserTest.java [hunk 1]
```diff
@@ -18,14 +21,121 @@ public void selfMethod() {
...
     }
 
     @Test
     public void staticMethod() {
         assertEquals(String.valueOf(User.getTotalUsersCount()), User.userCountStatus());
     }
+
+    private <T> void assertValueRoundTrip(Function<T, T> func, Function<T, String> toString, T value, String text) {
+        assertEquals(value, func.apply(value));
+        assertEquals(text, toString.apply(value));
```
Comment (ahouts) [inline tests/driver/src/test/java/UserTest.java:139]: Decided this was also probably a good idea to make sure that the value rust has is what we expect it to be.
For example, if we converted java booleans to rust bools incorrectly then from rust bools to java booleans incorrectly, we would wind up with the correct java value in the full trip test. This allows us to check what rust has in the middle.

File: src/convert/mod.rs [hunk 2]
```diff
@@ -89,7 +89,7 @@ macro_rules! jvalue_types {
             fn autobox(self, env: &JNIEnv<'env>) -> JObject<'env> {
                 env.call_static_method_unchecked(concat!("java/lang/", stringify!($boxed)),
                     (concat!("java/lang/", stringify!($boxed)), "valueOf", concat!(stringify!(($sig)), "Ljava/lang/", stringify!($boxed), ";")),
-                    JavaType::from_str(concat!("java/lang/", stringify!($boxed))).unwrap(),
+                    JavaType::from_str(concat!("Ljava/lang/", stringify!($boxed), ";")).unwrap(),
```
Comment (ahouts) [inline src/convert/mod.rs:92]: The issue ended up being here, the `Lpath/Object;` syntax was required.

**Comment by ahouts** [comment]
I believe this should be good to review again, I was able to add a test for a list of primitive types.

**Review by giovanniberti — APPROVED** [review]

**Pull request merged**